### PR TITLE
Reload - Add manual reload for vehicle weapons - continued

### PR DESCRIPTION
**Pull request by johnb432**
**When merged this pull request will:**
- Title. Continuation of #5827, credit to @PabstMirror.
- `FUNC(canSwapTurretMagazine)` has been altered slightly from its original form, but the functionality is the same.
- `FUNC(swapTurretMagazine)` has been significantly changed. Previously it removed all turret magazines, whereas now it uses the `"LoadMagazine"` action to change magazines.
- Added an icon to the action (icon is from base game).

### IMPORTANT

- [ ] If the contribution affects [the documentation](https://github.com/acemod/ACE3/tree/master/docs), please include your changes in this pull request so the documentation will appear on the [website](https://ace3.acemod.org/).
- [x] [Development Guidelines](https://ace3.acemod.org/wiki/development/) are read, understood and applied.
- [x] Title of this PR uses our standard template `Component - Add|Fix|Improve|Change|Make|Remove {changes}`.

### CODE REVIEW

**Comment by johnb432** [comment]
Instead of this being an ACE interaction, it could be made to trigger when pressing the reload key instead. Thoughts?

**Comment by Drofseh** [comment]
> Instead of this being an ACE interaction, it could be made to trigger when pressing the reload key instead. Thoughts?

Both? Both!

**Comment by johnb432** [comment]
> Both? Both!

Fair enough xD

On another note: Should there be a CBA setting enabling/disabling this feature?

**Comment by Drofseh** [comment]
I don't really think it needs a setting

**Comment by PabstMirror** [comment]
```
    class HMG_01;
    class HMG_static: HMG_01 {
        type = 1;
    };
```
will let the engine reload via R key

**Comment by johnb432** [comment]
> ```
>     class HMG_01;
>     class HMG_static: HMG_01 {
>         type = 1;
>     };
> ```
> 
> will let the engine reload via R key

Should that be integrated into this PR? As in use that for static weapons, weapons on vehicles or vehicles (I'm unsure if it applies to `CfgVehicles `or `CfgWeapons`)?
Or it could check if ` type = 1` is set for the weapon/vehicle in the `addUserActionEventHandler`?

What do you suggest doing with that config entry?

**Comment by PabstMirror** [comment]
We could just add `type = 1;` too all static weapons to make them reloadable via engine
I can't see any downside to this except that it's more config work

**Comment by johnb432** [comment]
> ```
>     class HMG_01;
>     class HMG_static: HMG_01 {
>         type = 1;
>     };
> ```
> 
> will let the engine reload via R key

Are you sure about this? I have just tried loading CBA and ACE and using any static weapon (all vanilla static weapons have `type = 1`).

**Comment by ampersand38** [comment]
Would [repacking magazines / relinking belts](https://github.com/ampersand38/reload-repack-turret-magazines/blob/main/addons/reload-repack-turret-magazines/fn_reloadTurret.sqf#L53) be in scope here?

**Comment by johnb432** [comment]
> Would [repacking magazines / relinking belts](https://github.com/ampersand38/reload-repack-turret-magazines/blob/main/addons/reload-repack-turret-magazines/fn_reloadTurret.sqf#L53) be in scope here?

I think it would be best as a independent ACE interaction, instead of happening automatically when reloading (if I understood your code correctly).
But I would absolutely love it to be a part of ACE.

**Comment by PabstMirror** [comment]
we've really put it off but https://github.com/acemod/ACE3/pull/9085
contains a lot of what we want for repacking/relinking

**Comment by Drofseh** [comment]
I don't know if 9085 covers it or not, but any packing/linking for vehicles will need to keep the empty magazines or ace rearm can fail with scripted loadouts.

**Comment by johnb432** [comment]
> I don't know if 9085 covers it or not, but any packing/linking for vehicles will need to keep the empty magazines or ace rearm can fail with scripted loadouts.

I was thinking if I would need to adjust this PR for that, but it doesn't look like it. The reload action only happens if there is some ammo available in the weapon (but not a full magazine).

**Comment by johnb432** [comment]
We make it so that this is disabled when CSW is enabled for vehicles, when we get around to merging 9085.

File: addons/reload/functions/fnc_canSwapTurretMagazine.sqf [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+private _turretPath = _vehicle unitTurret _unit;
+if (_turretPath in [[-1], []]) exitWith {false}; // skip driver / cargo
+if !(_vehicle turretLocal _turretPath) exitWith {false}; // just to be safe
+
+(weaponState [_vehicle, _turretPath]) params ["_weapon", "_muzzle", "", "_magazine", "_ammoCount", "_roundReloadPhase", "_magazineReloadPhase"];
+TRACE_5("",_weapon,_muzzle,_magazine,_ammoCount,typeOf _vehicle);
+
+if ((_weapon == "") || {_weapon != _muzzle}) exitWith {false}; // skip multi-muzzle (he/ap auto-cannons)
+if (_magazine == "") exitWith {false};
+if (_roundReloadPhase + _magazineReloadPhase != 0) exitWith {false}; // can't reload while already reloading
```
Comment (LinkIsGrim) [inline addons/reload/functions/fnc_canSwapTurretMagazine.sqf:31]:
```suggestion
if (_roundReloadPhase + _magazineReloadPhase != 0) exitWith {false}; // can't reload while already reloading or while shooting
```

**Review by LinkIsGrim — APPROVED** [review]

**Pull request merged**